### PR TITLE
fix: minimum 1 bit for constant binary decomposition

### PR DESCRIPTION
**Pull request by gbotrel**
# Description

Fixes an edge case introduced by #1228 ; thanks to @wuestholz for reporting.

### CODE REVIEW

**Comment by ivokub** [comment]
Suggested edit:
```diff
diff --git a/internal/regression_tests/issue1227/issue_1227_test.go b/internal/regression_tests/issue1227/issue_1227_test.go
index 3315f54d..a9342995 100644
--- a/internal/regression_tests/issue1227/issue_1227_test.go
+++ b/internal/regression_tests/issue1227/issue_1227_test.go
@@ -8,17 +8,22 @@ import (
 )
 
 type Circuit struct {
-	X frontend.Variable
+	constVal int
+	X        frontend.Variable
 }
 
 func (circuit *Circuit) Define(api frontend.API) error {
-	api.AssertIsLessOrEqual(1, circuit.X)
+	api.AssertIsLessOrEqual(circuit.constVal, circuit.X)
 	return nil
 }
 
 func TestConstantPath(t *testing.T) {
 	assert := test.NewAssert(t)
-	assert.CheckCircuit(&Circuit{},
+	assert.CheckCircuit(&Circuit{constVal: 1},
 		test.WithValidAssignment(&Circuit{X: 1}),   // 1 <= 1 --> true
 		test.WithInvalidAssignment(&Circuit{X: 0})) // 1 <= 0 --> false
+	// test edge case where constant is 0
+	assert.CheckCircuit(&Circuit{constVal: 0},
+		test.WithValidAssignment(&Circuit{X: 1}), // 0 <= 1 --> true
+		test.WithValidAssignment(&Circuit{X: 0})) // 0 <= 0 --> true
 }

```

**Review by ivokub — APPROVED** [review]
LGTM. Suggested edit for catching regressions in the future.
